src/components/KSelect/KSelect.vue

```diff
@@ -671,34 +512,38 @@ const triggerFocus = (evt: any, isToggled: Ref<boolean>):void => {
     return
   }
 
-  const inputElem = document.getElementById(selectTextId.value)
+  const inputElem = selectWrapperElement.value?.children[0] as HTMLInputElement
   if (!isToggled.value && inputElem) { // simulate click to trigger dropdown open
     inputElem.click()
   }
 }
 
 const onQueryChange = (query: string) => {
-  if (filterStr.value !== query) {
-    filterStr.value = query
-    emit('query-change', query)
+  if (filterQuery.value !== query) {
+    filterQuery.value = query
   }
 }
 
 const onInputFocus = (): void => {
   inputFocused.value = true
-  if (!initialFocusTriggered.value) {
-    initialFocusTriggered.value = true
-    emit('query-change', '')
-  }
+
+  emit('query-change', filterQuery.value)
 }
 
 const onInputBlur = (): void => {
   inputFocused.value = false
 }
 
+const onSelectWrapperClick = (event: Event): void => {
+  if (isDisabled.value || (event?.target as HTMLElement)?.dataset.propagateClicks === 'false') {
```

Comment:
nit: whole PR 

There are a ton of changes; can you add comments in the code please?